src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health) {
+    this.health = health || 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+  static updateAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(harbivore) {
+    if (!harbivore.hidden && harbivore instanceof Herbivore) {
```

Comment:
The check for 'harbivore.hidden' should be 'herbivore.hidden' after correcting the parameter name. Additionally, the order of conditions should be reversed to first check if 'herbivore' is an instance of Herbivore, which is a more fundamental check, before checking if it is hidden.